### ISSUE TITLE
Game files will not transfer into the "Drag & Drop Games Here" screen a second time

### ISSUE THREAD

**Issue by Rotgut-X** [OpenEmu/OpenEmu#3021]
Hi.  First, **thank you** if you can help me out.  I love OpenEmu for the Mac, it's pretty amazing.  I did not see this question asked so here goes.

Everything was going **GREAT** until this issue.  I had transferred dozens of games into several systems (NES, TurboGrafx-16, etc).  The games appeared and were running, all was perfect!  For organizational reasons, I then decided to delete games from the screen for one of the systems and kind of "start over" with transferring them — but now when go to transfer the game files back into the "Drag & Drop Games Here" screen like before — the games refuse to appear!  The Game Scanner in the bottom left corner appears to "scan" but the result is... no game!  I have tried renaming game file folders, I have tried restarting the Mac.  Nothing solves this issue.  Anyone?

Note: I am using OpenEmu experimental ver 2.0.4 (639-g606b287-Release) since I have added some CD-based systems like Saturn and Turbo Duo.

**Comment by clobber** [comment]
Please provide exact details on which game will not import.

**Comment by Rotgut-X** [comment]
Every single one of the games for each of the CD-based systems behave the exact same way: OpenEmu only allows you to transfer game files/folders **one time** to an OpenEmu "Consoles" screen (ie; Sega Saturn).  They open and play perfectly!  BUT - - the minute you delete a game - any game - from the OpenEmu "Consoles" screen (ie; Sega Saturn)...  and attempt to drag/drop it back onto the screen after that... **it will not appear.**  

The non-CD based systems do not behave this way.  I can delete a game and then re-transfer it and it does indeed appear.

**Comment by clobber** [comment]
Cannot replicate this - I can import and delete Saturn Bomberman over and over.

**Comment by Rotgut-X** [comment]
Hmm.  Interesting. Is there a place where misc. files are stored?  Maybe I can delete those files delete all files associated w OpenEmu and/or do a complete uninstall?  And then do a re-install from scratch?  Keep in mind I am using a Mac.  (I am more familiar with doing emulation on PC)

**Comment by Rotgut-X** [comment]
Okay, so I kept fooling around with Preferences and I think I've figgured the glitch: if you have the "copy files to Game Library when adding to library" box checked this happens...  when you uncheck that box it lets me add games and things seem ok...  very odd

**Comment by danomatika** [comment]
I'm seeing the same thing. I'm also using the "Copy files to Gane Library" preference.

Mini story:

I set up OpenEMU and added a whole library of games. Then I decided to clean some things up and removed a few, selecting the "Move to Trash" button. I accidentally removed a couple I wanted to keep, so I took them out of the Trash and dragged them onto OpenEMU. Nothing happened.

Ok, so I tried removing and reading all the files. Now the library list is completely empty and I can no longer add anything (or at least it seems like I can't). Adding the files back to the rom subfolder in the Game Library does nothing as well.

Ok, so maybe the db or preferences need to be refreshed. I moved the library, created a new one, and tried dragging in the folder of games again. Nothing, the list is still empty. I nuked the files in `~/Library/Application Support/OpenEMU` and still no change.

**Comment by clobber** [comment]
Just an FYI: "Copy files to Game Library folder when adding to library" is the default option - cannot replicate this behavior unless there is an precise step or condition I'm missing. This is the first we've heard of this problem, despite having OpenEmu 2.x out for over a year.

Usually, the only time a game cannot import is if the files already exist in the Game Library folder (OpenEmu will not allow the import of duplicates).

Please provide exact steps (very precise to the letter) to replicate this problem including a specific game, so we can all check it out. Also include details of your Preferences settings if you deviated from any of the defaults.

**Comment by danomatika** [comment]
>This is the first we've heard of this problem, despite having OpenEmu 2.x out for over a year.

I'll assume you're glad for the follow up bug report, then. :) Even if I'm doing it the _wrong way_.

>Usually, the only time a game cannot import is if the files already exist in the Game Library folder (OpenEmu will not allow the import of duplicates).

As I mentioned before, I've tried starting over with a new library location but OpenEMU still seems to know which files it's already imported into the old library, so this seems related.

>Please provide exact steps (very precise to the letter) to replicate this problem including a specific game, so we can all check it out. Also include details of your Preferences settings if you deviated from any of the defaults.

I can't replicate it exactly as I can't get back to a state where importing works. Is there a list of preferences  (stuff in `~/Library`, hidden home folder, etc) I can delete to get back to a fresh install? Then I could double check & re-write the first paragraph from my last point in an ordered list for . :)

**Comment by danomatika** [comment]
So far, I'm unable to completely wipe OpenEMU so the issue persists. It seems like there is a preferences / db cache left one _somewhere_ that keeps getting picked up, even after running the OpenEMUWiper script. Preference options like "Keep games organized" persist between "wipes." This may be a separate issue though.

**Comment by clobber** [comment]
For both of you, please:

Open up Terminal.app and run the following command:
`defaults read.org.openemu.OpenEmu > ~/Downloads/openemuprefs.txt`

Then go to your Downloads folder upload openemuprefs.txt file here.

**Comment by Rotgut-X** [comment]
I don't know what Terminal.app is (sorry)...?

**Comment by clobber** [comment]
There's this thing called [Spotlight search](https://support.apple.com/en-us/HT204014) (magnifying glass in the upper right corner on your screen - it's how one finds files and apps on their Mac). You click it and type Terminal and press enter. Then copy and paste the line above to get the output.

**Comment by Rotgut-X** [comment]
[openemuprefs.txt](https://github.com/OpenEmu/OpenEmu/files/722573/openemuprefs.txt)

**Comment by danomatika** [comment]
Ah ok, OpenEmu is using NSUserDefaults, so that answers my question as per Preferences. Sorry I didn't think of it earlier.

[openemuprefs.txt](https://github.com/OpenEmu/OpenEmu/files/722610/openemuprefs.txt)

**Comment by cyco** [comment]
Hey, thanks for posting your preferences. Can you check your rom directories and see if the deleted files still exist there? Also check the "unsorted" directory. If you find the files just move them somewhere else and try another import.

**Comment by danomatika** [comment]
The files were in the `Super Nintendo (SNES)` folder as I tried moving/removing them the other day a few times. The `unsorted` & `Automatically Import` folders are empty.

I went ahead cleared the SNES folder, started OpenEMU, and tried dragging them back in from the Desktop. The games list remains empty and no files are copied into the library rooms folders. I'm not seeing any error prints in Console either.

**Comment by clobber** [comment]
>I went ahead and cleared the SNES folder

Huh? Are you just manually deleting and copying files to sub folders in your Game Library without first deleting or importing from the Library using the OpenEmu UI? If so, this may be why you are having problems.

**Comment by danomatika** [comment]
When the UI shows nothing, I can't do it through the UI now can I?

**Comment by danomatika** [comment]
Also, please keep in mind that I don't have the background knowledge to OpenEMU development that y'all have. If I am doing things the _wrong way_ it's because I don't know what the _right way_ is. I'm jumping into this cold, only having downloaded OpenEMU last week, and ran into this issue.

Where is the list of already imported games stored? In the library database itself or somewhere else?

**Comment by clobber** [comment]
I suggest:

1. Delete OpenEmu.app
2. Delete the OpenEmu folder in ~/Library/Application Support
3. Delete Game Library in ~/Documents
4. Run `defaults delete org.openemu.OpenEmu` in Terminal
5. Restart your computer for good measure
6. Redownload OpenEmu 2.0.4 from the [website](http://openemu.org/) and install
7. Open the app and go through the setup process, don't change anything in Prefs including your Game Library location
8. Import a single unzipped ROM (such as SNES, .sfc/.smc file)

**Comment by danomatika** [comment]
1. ran the following:

~~~
rm -rf /Applications/OpenEmu.app
rm -rf ~/Documents/Game\ Library
rm -rf ~/Library/Application\ Support/OpenEmu
defaults delete org.openemu.OpenEmu
~~~

2. restarted
3. downloaded & installed OpenEMU 2.0.4 release
4. ran setup and kept all defaults
5. imported ActRaiser (US).smc

Game was imported and plays fine. I am also able to remove it from the library and re-add it with no problems.

Further, some notes from my first install last week:

* the overall library I have consists of zipped games which did import the first time & were playable
* I did some reorganization and threw some unsupported console rooms into the library `roms` subfolder: `MSX`, `NeoGeo`

Interestingly, the zip SNES games I have do not import with this fresh install.

**Comment by clobber** [comment]
They have to be single file per zip, not containing multiple games or other files.

**Comment by danomatika** [comment]
My library is pretty old at this point, 10+ years at this point. Extracting one of the zips generates a .zip.cpzg which then extracts into a .zip again and round and round, so there may be issues with my files. They are single roms per zip, AFAICT.

**Comment by clobber** [comment]
Hrm that sounds like either a problem with the decompression software or a bad zip. Try [The Unarchiver](https://itunes.apple.com/us/app/the-unarchiver/id425424353?mt=12), otherwise if that fails just download new ROMs preferably from a "no-intro" set (google).

**Comment by danomatika** [comment]
Unarchiver chokes on them too. The weird thing is that OpenEMU imported them the first time just fine and I was able to play. It was only after I cleared and tried to import that I started running into issues. I also remember seeing OpenEMU zip archive errors on the Console on Sunday, but I can't find them now (logs must have been purged already).

I'd chalk this up to a system issue with unarchiving malformed/old zip archives. I'm pulling fresh rooms and rebuilding my library.

**Comment by clobber** [comment]
OpenEmu uses the same framework that The Unarchiver utilizes (XADMaster), so it is unlikely the same game originally imported that The Unarchiver cannot extract. Anyway, yeah getting new ROMs is a good idea.

Your issue seems unrelated to the one @Rotgut-X originally reported regarding **CD-based games**, which is probably due to files lingering in the Game Library (not being moved to trash).

**Comment by danomatika** [comment]
I can provide a link to one of the files, if needed.

**Comment by Rotgut-X** [comment]
So I have what you may deem a very, very dumb question.  Where exactly is the Game Library located?  There are no games in the Application folder as described.

I have all of my roms in a folder not associated with OpenEmu.  Yes, I did a search on my Mac to see where the Game Library folder might reside but cam up with nothing.

**Comment by Ryno9828** [comment]
I am having the exact same problem as this person. the ROMs that i deleted, even with a fresh install of the program, will not import back into OpenEmu.

**Comment by clobber** [comment]
This issue is addressed by commits 5f3dbba297f5fe41ee3c0d278d3c72e1b973771d & 774496ff4b84084abd68d8bf5e3736041536fd89

**Comment by KhaledElAnsari** [comment]
I had the same issue after I removed a game from the library without moving the files to trash

I solved it by going to the *Game Library* file and remove a folder called `unsorted` then I imported the game again and it worked!

**Comment by SageCord** [comment]
I had an issue similar to this but found a better solution. When trying to add the games, make sure they don't end in .rar. I was trying to use NDS so I had to change them to .nds

**Comment by SageCord** [comment]
Wait. No. do not do this. It does not work.
Does anybody have any suggestions for me?
